spec/routing/media_routing_spec.rb

```diff
@@ -6,23 +6,111 @@
       'media#download', id: 'oo000oo0000', file_name: 'aa666aa1234', format: 'mp4')
   end
 
-  it 'download: filename with hyphen' do
-    expect(get: '/media/oo000oo0000/aa666aa1234-a.mp4').to route_to(
-      'media#download', id: 'oo000oo0000', file_name: 'aa666aa1234-a', format: 'mp4')
-  end
+  context '#download: filename with' do
+    it 'hyphen' do
+      expect(get: '/media/oo000oo0000/aa666aa1234-a.mp4').to route_to(
+        'media#download', id: 'oo000oo0000', file_name: 'aa666aa1234-a', format: 'mp4')
+    end
+
+    it 'chars not requiring URI escaping' do
+      # filename = "(no_escape_needed):;=&$*.-_+!,~'" # can't use dot here, see applicable routing constraint
+      filename = "(no_escape_needed):;=&$*-_+!,~'"
+      expect(get: "/media/oo000oo0000/#{filename}.mp4").to route_to(
+        'media#download', id: 'oo000oo0000', file_name: filename, format: 'mp4')
+    end
+
+    it 'some chars requiring URI escaping' do
+      filename = 'escape_needed {} @#^ %|"`'
+      expect(get: "/media/oo000oo0000/#{URI.escape(filename)}.mp4").to route_to(
+        'media#download', id: 'oo000oo0000', file_name: filename, format: 'mp4')
+    end
+
+    it 'square brackets must be url escaped' do
+      skip("problem writing test:  the route doesn't seem to match on brackets, even if they're escaped")
```

Comment:
why would this be different when the code is so similar to file#show ?   Maybe we can pair and poke around together?

also - it works for #stream, but not #download?  wacko.